### PR TITLE
(De)serialize inputs connectee names.

### PR DESCRIPTION
**Pull request by chrisdembia**
This PR allows one to serialize and deserialize input connectee names, as is done for connectors. The major issue I ran into was handling the `isList` member variable of AbstractConnector, which is not restored across serialization. The way I handled this for now was to store the `isList` setting in the `_inputsTable`, which is then passed onto the AbstractInput after deserialization (through the `AbstractConnector::initialize()` method). However, I hope to fix that issue in #1009.

I also did a bunch of miscellaneous cleanup.

This PR builds on top of @klshrinidhi's TableSource PR #994, and should not be merged until after that PR is merged.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/1012)

### CODE REVIEW

File: OpenSim/Common/Component.cpp [hunk 1]
```diff
@@ -173,8 +176,27 @@ void Component::finalizeFromProperties()
     // TODO if we implement custom copy constructor and assignment methods,
     // then this could be moved there (and then Output could take owner as an
     // argument to its constructor).
-    for (auto& it : _inputsTable) {
-        it.second->setOwner(*this);
+    if (getProperty_connectors().size() != _connectorsTable.size()) {
+        std::stringstream msg;
+        msg << "Expected " << _connectorsTable.size()
+            << " Connectors but got " << getProperty_connectors().size() << ".";
+        OPENSIM_THROW_FRMOBJ(Exception, msg.str());
```
Comment (chrisdembia) [inline OpenSim/Common/Component.cpp]: This exception, and the one below, would go away if we do #1009.

File: OpenSim/Common/Component.h [hunk 2]
```diff
@@ -2042,22 +2043,10 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
...
-        _connectorsTable.clear();
-        for (int ix = 0; ix < getProperty_connectors().size(); ++ix){
-            AbstractConnector& connector = upd_connectors(ix);
-            connector.setOwner(*this);
-            _connectorsTable[connector.getName()] = ix;
-        }
-        
-        for (auto& it : _inputsTable) {
-            it.second->setOwner(*this);
-    }
```
Comment (chrisdembia) [inline OpenSim/Common/Component.h]: This code moved to `finalizeFromProperties()`.

File: OpenSim/Common/Component.h [hunk 3]
```diff
@@ -2395,94 +2384,98 @@ template<class T>
 void Input<T>::connect(const AbstractOutput& output,
                        const std::string& annotation) {
     const auto* outT = dynamic_cast<const Output<T>*>(&output);
```
Comment (chrisdembia) [inline OpenSim/Common/Component.h]: The rest of the changes in this file are simply rearranging code and renaming variables. The diff really makes this section hard to read, but I really didn't make substantial changes here.

File: OpenSim/Common/ComponentConnector.h [hunk 4]
```diff
@@ -156,17 +158,15 @@ class OSIMCOMMON_API AbstractConnector : public Object {
...
 
     /** Set connectee name of a connectee among a list of connectees. This
     function is used if this connector is a list connector.                   */
     void setConnecteeName(const std::string& name, unsigned ix) {
-        OPENSIM_THROW_IF(ix >= getNumConnectees(),
-                         IndexOutOfRange,
-                         ix, 0, static_cast<unsigned>(getNumConnectees() - 1));
+        using SimTK::isIndexInRange;
+        SimTK_INDEXCHECK_ALWAYS(ix, getNumConnectees(),
+                                "AbstractInput::setConnecteeName()");
```
Comment (chrisdembia) [inline OpenSim/Common/ComponentConnector.h]: This macro handles the max index being 0.

File: OpenSim/Common/ComponentConnector.h [hunk 5]
```diff
@@ -323,15 +344,52 @@ class Connector : public AbstractConnector {
...
+   (presumably a component that averages its inputs) has an output named
+   `output`, and we are annotating this output as `knee_joint_center`.
+ - `/leg_model/soleus/activation`: This connectee name uses the absolute path
+   to component `soleus`, which has an output named `activation`.
+
+List inputs can contain multiple entries in its `connectee_name`, with the
+entries separated by a space. For example:
+@verbatim
+../marker_data/column:left_ankle ../marker_data/column:right_ankle ../averager/output(knee_joint_center)
+@endverbatim
```
Comment (chrisdembia) [inline OpenSim/Common/ComponentConnector.h]: The above might change according to suggestions in #1009.

**Comment by chrisdembia** [comment]
The last commit fixes an issue I ran into an issue where we had a model committed to the repo (version 30503) in which Station's connector was named "reference_frame," which is now named "parent_frame." Based on discussion in the dev meeting, I reverted the model back to the 30000 version, but preserved the contents of the model (that is, the CoordinateLimitForce and residual/reserve forces that I think @aymanhab added at the same time the model was updated to version 30503). 

I also enhanced some checks/exceptions to catch exactly when the connector/input properties do not match those in the component.

**Comment by chrisdembia** [comment]
FYI this is ready for review.

**Comment by chrisdembia** [comment]
I just fixed a merge issue. This PR is ready to be reviewed.

**Comment by aseth1** [comment]
*[OpenSim/Common/ComponentConnector.h, line 202 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKo9j-V3sXYxQPi_HtU:-KKo9j-WF63WG7BiQEg8:990454833) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L202)):*
> ```objc
> protected:
>     const Component& getOwner() const { return _owner.getRef(); }
>     void initialize(const Component& o, bool isList = false) {
> ```

I understand why the you rename the method now that you also have to provide the isList flag. Perhaps we can think of something more descriptive than "initialize" (after this PR would be fine)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/ComponentConnector.h, line 234 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKoBCaEDPXPaOeJbgy_:-KKoBCaEDPXPaOeJbgya:492882487) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L234)):*
> ```objc
>         }
>     }
>     SimTK::Stage connectAtStage = SimTK::Stage::Topology;
> ```

If you want this to be initialized to invalid Stage you could do that. If you want to it to be lowest stage possible to have a "model" then this should be Stage::Model.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/Test/TestTableSource.osim, line 6 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKuyHlxjv1mdXfYUgpW:-KKuyHlyqQO2cax5B4al:-479323665) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/Test/TestTableSource.osim#L6)):*
> ```
>           <components>
>             <TableSource__Vec3_ name="tablesource">
>               <filename>aStaticTrial_2.c3d</filename>
> ```

#994 makes this .trc right?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/ComponentConnector.h, line 826 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKunYniAwNqZPvD2zBn:-KKunYniAwNqZPvD2zBo:-625958104) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L826)):*
> ```objc
>     /** @}                                                               */ \
>     /** @cond                                                            */ \
>     int _input_TODO_##iname {                                               \
> ```

Why the TODO in the function name?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/Test/testComponentInterface.cpp, line 1057 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKuo9w8fmK9k6CI19Mg:-KKuo9w8fmK9k6CI19Mh:-1712610568) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/Test/testComponentInterface.cpp#L1057)):*
> ```cpp
>     using namespace SimTK;
> 
> #ifdef WITH_BTK
> ```

The #ifdef will be moved with @klshrinidhi 's changes #994 to use a trc file for TableSourceVec3? I think that would be better than a test dependent on having BTK

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/ComponentConnector.h, line 169 \[r2\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKQW3ckn15-uHyB5Hai-r2-169:-KKv15p5QbcCS_hhzh6y:1765920889) ([raw file](https://github.com/opensim-org/opensim-core/blob/b2ca3b81751c31f192a58b6074a2101d06121d00/OpenSim/Common/ComponentConnector.h#L169)):*
<details><summary><i>Previously, chrisdembia (Christopher Dembia) wrote…</i></summary>
> This macro handles the max index being 0.
</details>
Should we remove Exception::IndexOutOfRange in the near future. Would be good to reduce redundancy with SimTK as much as possible.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/ComponentConnector.h, line 133 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv2YTq9lJAkgczs8eO:-KKv2YTq9lJAkgczs8eP:-1531853111) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/ComponentConnector.h#L133)):*
> ```objc
>     virtual bool isConnected() const = 0;
>     
>     /** The number of desired connectees. This is 1 for a non-list connector. */
> ```

Desired or available number of connectees? This is the number of slots to fill to satisfy the Connector and not sure desire has anything to do with it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/Component.cpp, line 264 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv3hZH9wEUOXgbIKdk:-KKv3hZI2cvtsdGuOU4r:-873621033) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/Component.cpp#L264)):*
> ```cpp
>     }
> 
>     // rebuilding the connectors table, which was emptied by clearStateAllocations
> ```

this comment is only true if reset() was called which you removed. Why not reset before connecting to ensure no stale connections?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
*[OpenSim/Common/Component.cpp, line 747 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv49lDrJ_s00T9ejx1:-KKv49lEhV_LgWtsRyzp:1780540164) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/Component.cpp#L747)):*
> ```cpp
> }
> 
> const AbstractInput* Component::findInput(const std::string& name) const
> ```

Not for this PR, but there is a lot duplication in how we have to find Connectors, Inputs and Outputs and it would be great to implement in one place for all three types of "attributes" of the Component

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
Reviewed 4 of 7 files at r2, 2 of 2 files at r3, 3 of 4 files at r5.
Review status: all files reviewed at latest revision, 12 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by aseth1** [comment]
Done my review. No major issues I think except for the removal of `reset()` at the beginning of `Component::connect()`. That was a safeguard against having stale (satisfied) connections laying around, which may longer be valid. We can discuss tomorrow.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-:-KKuyeaCcbxHinmhwfey:71816822)*

**Comment by chrisdembia** [comment]
@aseth1 thanks for the review. Let me know if you still think removing `reset()` is an issue based on our discussion. `connect()` still disconnects everything first.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-:-KKw5ID_kFgNOueHjfu2:-2008810524)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 133 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv2YTq9lJAkgczs8eO:-KKw5pJzDotaWYg3mJBe:-1968659941) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/ComponentConnector.h#L133)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> Desired or available number of connectees? This is the number of slots to fill to satisfy the Connector and not sure desire has anything to do with it.
</details>
I like `number of slots to fill`. That is correct.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 169 \[r2\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKQW3ckn15-uHyB5Hai-r2-169:-KKw5wj-JLXOvjHb6Izg:534050790) ([raw file](https://github.com/opensim-org/opensim-core/blob/b2ca3b81751c31f192a58b6074a2101d06121d00/OpenSim/Common/ComponentConnector.h#L169)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> Should we remove Exception::IndexOutOfRange in the near future. Would be good to reduce redundancy with SimTK as much as possible.
</details>
Simbody's exceptions are written for use by Simbody, and are not themselves an "exception library." For example, many simbody exceptions print a message saying "This is a bug, submit a bug report at github.com/simbody/simbody." So we should have our own suite of exceptions instead of using simbody's. Or, rewrite simbody's exceptions to be generic "exception library" so that we could insert our own messages like "submit a but report to the simtk forum."

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 202 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKo9j-V3sXYxQPi_HtU:-KKw6HL8l44D9iIMtWGx:-1460748487) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L202)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> I understand why the you rename the method now that you also have to provide the isList flag. Perhaps we can think of something more descriptive than "initialize" (after this PR would be fine)
</details>
Good point; initialize is nearly meaningless. How about `syncWithComponent()`?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 826 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKunYniAwNqZPvD2zBn:-KKw6UY0YPSEb-YBik_D:630316746) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L826)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> Why the TODO in the function name?
</details>
Oops...thanks for catching that. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/Test/testComponentInterface.cpp, line 1057 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKuo9w8fmK9k6CI19Mg:-KKw6c04AqtyjPOM8gu5:-499146009) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/Test/testComponentInterface.cpp#L1057)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> The #ifdef will be moved with @klshrinidhi 's changes #994 to use a trc file for TableSourceVec3? I think that would be better than a test dependent on having BTK
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/Test/TestTableSource.osim, line 6 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKuyHlxjv1mdXfYUgpW:-KKw6eoYc57YM_OuW4Z9:-499146009) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/Test/TestTableSource.osim#L6)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> #994 makes this .trc right?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/Component.cpp, line 747 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv49lDrJ_s00T9ejx1:-KKyL5SB5ccaGEREjVqm:-1936361515) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/Component.cpp#L747)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> Not for this PR, but there is a lot duplication in how we have to find Connectors, Inputs and Outputs and it would be great to implement in one place for all three types of "attributes" of the Component
</details>
I created #1048 for this. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/Component.cpp, line 264 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv3hZH9wEUOXgbIKdk:-KKykhMQTC8ebjab8nGK:182444783) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/Component.cpp#L264)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> this comment is only true if reset() was called which you removed. Why not reset before connecting to ensure no stale connections?
</details>
Discussed at the top.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/Component.cpp, line 264 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv3hZH9wEUOXgbIKdk:-KKykjXKvsGd-ecc_m7G:-499146009) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/Component.cpp#L264)):*
<details><summary><i>Previously, chrisdembia (Christopher Dembia) wrote…</i></summary>
> Discussed at the top.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 826 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKunYniAwNqZPvD2zBn:-KKykzp49oyKiXR2o_-s:-499146009) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L826)):*
<details><summary><i>Previously, chrisdembia (Christopher Dembia) wrote…</i></summary>
> Oops...thanks for catching that. 
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 234 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKoBCaEDPXPaOeJbgy_:-KKyl0RfcdWDbB4kRSNL:-499146009) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L234)):*
<details><summary><i>Previously, aseth1 (Ajay Seth) wrote…</i></summary>
> If you want this to be initialized to invalid Stage you could do that. If you want to it to be lowest stage possible to have a "model" then this should be Stage::Model.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 202 \[r3\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKo9j-V3sXYxQPi_HtU:-KKyl296sPKHlTmazz88:-1083771322) ([raw file](https://github.com/opensim-org/opensim-core/blob/645dd34db7c6c7841905049aceb9d64ee011841b/OpenSim/Common/ComponentConnector.h#L202)):*
<details><summary><i>Previously, chrisdembia (Christopher Dembia) wrote…</i></summary>
> Good point; initialize is nearly meaningless. How about `syncWithComponent()`?
</details>
Changed to `restoreMembers()`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
*[OpenSim/Common/ComponentConnector.h, line 133 \[r5\]](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-KKv2YTq9lJAkgczs8eO:-KKyl5TBf0gOlyUb3S8W:-499146009) ([raw file](https://github.com/opensim-org/opensim-core/blob/51f070c22a5c369eda32aab68a71b51c3cc68ee4/OpenSim/Common/ComponentConnector.h#L133)):*
<details><summary><i>Previously, chrisdembia (Christopher Dembia) wrote…</i></summary>
> I like `number of slots to fill`. That is correct.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
@aseth1 ready for review again.

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-:-KKylIHQVwQzlME0YY3_:-1399807065)*

**Comment by chrisdembia** [comment]
Great. We can wait to merge this after your geometry PR gets merged; it will surely have conflicts with that PR

**Comment by aseth1** [comment]
:lgtm:

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012#-:-KKypKaTR6cO7FM80Qk6:1417088721)*

**Comment by aseth1** [comment]
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/opensim-org/opensim-core/1012)*

**Comment by chrisdembia** [comment]
@aseth1 tests pass. Beware that this branch conflicts with the geometry branch. If you want I can resolve those conflicts for you (ie, merge with master) and push the fixes to the geometry branch.

We could also wait to merge this after the geometry stuff gets merged.

**Comment by aseth1** [comment]
@chrisdembia are you still resolving merge conflicts?

**Comment by chrisdembia** [comment]
The issue is not merge conflicts. The issue is that checks for equality before/after serialization fail (I think unless one calls `finalizeFromProperties()` before checking for equality?). It may be best to  start working on the alternative scheme for storing the connectee_name for inputs/connectors that we discussed before the workshop. I haven't been working on this.

**Comment by aseth1** [comment]
@chrisdembia thanks for letting me know. I recommend making the `finalizeFromProperties()` call in order to make the comparison and pass the tests. The alternate scheme can be addressed in forthcoming PR(s). It would be good to get this PR in with minimal additional changes.

**Comment by chrisdembia** [comment]
Sounds good. I'll try that.

**Comment by aseth1** [comment]
>Sounds good. I'll try that

@chrisdembia did you get a chance to give it a try?

**Comment by chrisdembia** [comment]
> @chrisdembia did you get a chance to give it a try?

Yes; it ended up being more complicated than I thought.

**Comment by chrisdembia** [comment]
I'll work on this when I get back.

**Pull request closed without merging**